File: scripts/gen-rgb-css.py
RES = 360
COLORS = [
    "#fd004c",
    "#fe9000",
    "#fff020",
    "#3edf4b",
    "#3363ff",
    "#b102b7",
    "#fd004c",
]

with open("server/interface/timer_popup/rgb.css", "w") as f:
    step = (360/RES)
    
    f.write("@keyframes rgb {\n")

    for i in range(RES + 1):
        deg = round(i * step, 3)
        per = round(deg / RES * 100, 3)
        f.write(f"\t{per:>6}% {{ background: conic-gradient(")

        s = f"from {str(deg):>4}deg, "
        for color in COLORS:
            s += f"{color}, "            
        s = s[:-2]

        f.write(f"{s}) }}\n")

    f.write("}\n")

File: scripts/gen-rgb-css.js
const fs = require("fs");
process.chdir("interface");

const RES = 360
const COLORS = [
    "#fd004c",
    "#fe9000",
    "#fff020",
    "#3edf4b",
    "#3363ff",
    "#b102b7",
    "#fd004c",
]
const step = 360 / RES;

var data = "@keyframes rgb {\n";
for (let i = 0; i < RES + 1; i++) {
    const deg = (i * step).toFixed(3);
    const per = (deg / RES * 100).toFixed(3);
    data += `\t${per.toString().padStart(6, " ")}% { background: conic-gradient(`

    var s = `from ${deg.toString().padStart(6, " ").toString()}deg, `
    for (let color of COLORS) {
        s += `${color}, `
    }
    s = s.slice(0, -2)

    data += s + `); }\n`
}
data += "}\n"


fs.writeFileSync("rgb.css", data);

console.log(`${process.cwd()}\\rgb.css created`);